EasyPost.Tests/BetaFeaturesTests/ParametersTests/ParametersTest.cs

```diff
@@ -0,0 +1,152 @@
+            parameters = new Shipments.Create
+            {
+                IsReturn = false,
+                ToAddress = newAddressCreationParameters,
+            };
+
+            // No errors here confirm that the dictionary was serialized to the expected schema
+            dictionary = parameters.ToDictionary();
+            shipmentData = dictionary["shipment"] as Dictionary<string, object>;
+            addressData = shipmentData["to_address"] as Dictionary<string, object>;
+
+            // The value of "street1" should be the value of "streetB" via the Address.Create parameter object
+            Assert.Equal(streetB, addressData["street1"]);
+        }
+
+        [Fact]
+        [Testing.Logic]
+        public void TestTopLevelVersusNestedParameters()
+        {
+            // This test proves that we can reuse the Address.Create parameter object,
+            // with its serialization logic adapting to whether it is
+            // a top-level parameter object or a nested parameter object within another parameter object.
+
+            const string street = "388 Townsend St";
+
+            Addresses.Create addressCreationParameters = new Addresses.Create
+            {
+                Street1 = street,
+            };
+
+            // Using the Address.Create parameter object as a top-level parameter object
+            var dictionary = addressCreationParameters.ToDictionary();
+
+            // Path to "street1" should be dictionary["address"]["street1"]
+            var addressData = dictionary["address"] as Dictionary<string, object>;
+            Assert.Equal(street, addressData["street1"]);
+
+            // Using the Address.Create parameter object as a nested parameter object
+            var shipmentCreationParameters = new Shipments.Create
+            {
+                IsReturn = false,
+                ToAddress = addressCreationParameters,
+            };
+            dictionary = shipmentCreationParameters.ToDictionary();
+
+            // Path to "street1" should be dictionary["shipment"]["to_address"]["street1"]
+            var shipmentData = dictionary["shipment"] as Dictionary<string, object>;
+            var toAddressData = shipmentData["to_address"] as Dictionary<string, object>;
+            Assert.Equal(street, toAddressData["street1"]);
+
+            // Notice how the paths to "street1" are different depending on whether the Address.Create parameter object
+            // The schema for an address creation API call contains all address data wrapped in an "address" key (excluding irrelevant "verify" and "verify_strict" keys)
+            // The schema for a shipment creation API call does not contain this "address" key, with all address data instead wrapped inside a "to_address" key
+            // Behind the scenes, the Address.Create parameter object is the same, but the serialization path for each property (parameter) adapts to the context in which it is used
```

Comment:
This should also be in the docstring of the test, not buried at the bottom of a function. I'd expect this to be a paragraph after the intro line to the test in the docstring.